src/uct/ib/base/ib_md.c

```diff
@@ -310,28 +311,40 @@ static ucs_status_t uct_ib_md_query(uct_md_h uct_md, uct_md_attr_t *md_attr)
 static void uct_ib_md_print_mem_reg_err_msg(ucs_log_level_t level, void *address,
                                             size_t length, uint64_t access_flags)
 {
-    char msg[400] = {0};
+    int err = errno; /* save errno for case if it is overwritten */
+    ucs_string_buffer_t msg;
     struct rlimit limit_info;
     size_t page_size;
     size_t unused;
-    ucs_snprintf_zero(msg, sizeof(msg),
-                      "%s(address=%p, length=%zu, access=0x%lx) failed: %m",
-                      ibv_reg_mr_func_name, address, length, access_flags);
+
+    ucs_string_buffer_init(&msg);
+
+    ucs_string_buffer_appendf(&msg,
+                              "%s(address=%p, length=%zu, access=0x%lx) failed: %m",
+                              ibv_reg_mr_func_name, address, length, access_flags);
     /* Check the value of the max locked memory which is set on the system
      * (ulimit -l) */
     if (!getrlimit(RLIMIT_MEMLOCK, &limit_info) &&
         (limit_info.rlim_cur != RLIM_INFINITY)) {
-        ucs_snprintf_zero(msg + strlen(msg), sizeof(msg) - strlen(msg),
-                          ". Please set max locked memory (ulimit -l) to 'unlimited' "
-                          "(current: %llu kbytes)", limit_info.rlim_cur / UCS_KBYTE);
-    }
-    ucs_get_mem_page_size(address, length, &unused, &page_size);
-    if (page_size != ucs_get_page_size()) {
-        ucs_snprintf_zero(msg + strlen(msg), sizeof(msg) - strlen(msg),
-                          ". Application is using HUGE pages. "
-                          "Please set variable RDMAV_HUGEPAGES_SAFE=1");
+        ucs_string_buffer_appendf(&msg,
```

Comment:
pls check which errno code is generated if ulimit is limited and check it here (before adding the hint to set mac locked mem)

Reply:
done